tests/test_utils.py

```diff
@@ -634,7 +634,7 @@ def test_fetch_directory_html(include_dir_heading,       # type: bool
         out_dir = stack.enter_context(tempfile.TemporaryDirectory())
         out_files = fetch_directory(f"{tmp_host}/dir/", out_dir)
         expect_files = filter(lambda _f: _f.startswith("dir/"), test_http_dir_files)
-        expect_files = [os.path.join(out_dir, file.split("/", 1)[-1]) for file in expect_files]
+        expect_files = [os.path.join(out_dir, file) for file in expect_files]
```

Comment:
What happens now in this case if the fetched directory was `f"{tmp_host}/` instead? 
Since there is no directory/path part in the URL, does that cause an issue with inconsistent results of directory nesting?

Reply:
My understanding is that it would behave like the other comment below, `{out_dir}/test.com/`

I'll test it to make sure.